lib-ee/emqx_ee_connector/test/ee_connector_clickhouse_SUITE.erl

```diff
@@ -0,0 +1,199 @@
+% %%--------------------------------------------------------------------
+% %% Copyright (c) 2020-2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+% %%
+% %% Licensed under the Apache License, Version 2.0 (the "License");
+% %% you may not use this file except in compliance with the License.
+% %% You may obtain a copy of the License at
+% %% http://www.apache.org/licenses/LICENSE-2.0
+% %%
+% %% Unless required by applicable law or agreed to in writing, software
+% %% distributed under the License is distributed on an "AS IS" BASIS,
+% %% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% %% See the License for the specific language governing permissions and
+% %% limitations under the License.
+% %%--------------------------------------------------------------------
+
+-module(ee_connector_clickhouse_SUITE).
+
+-compile(nowarn_export_all).
+-compile(export_all).
+
+-include("emqx_connector.hrl").
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("emqx/include/emqx.hrl").
+-include_lib("stdlib/include/assert.hrl").
+
+-define(CLICKHOUSE_HOST, "clickhouse").
+-define(CLICKHOUSE_RESOURCE_MOD, emqx_ee_connector_clickhouse).
+
+%% This test SUITE requires a running clickhouse instance. If you don't want to
+%% bring up the whole CI infrastuctucture with the `scripts/ct/run.sh` script
+%% you can create a clickhouse instance with the following command (execute it
+%% from root of the EMQX directory.). You also need to set ?CLICKHOUSE_HOST and
+%% ?CLICKHOUSE_PORT to appropriate values.
+%%
+%% docker run -d -p 18123:8123 -p19000:9000 --name some-clickhouse-server --ulimit nofile=262144:262144 -v "`pwd`/.ci/docker-compose-file/clickhouse/users.xml:/etc/clickhouse-server/users.xml" -v "`pwd`/.ci/docker-compose-file/clickhouse/config.xml:/etc/clickhouse-server/config.xml" clickhouse/clickhouse-server
+
+all() ->
+    emqx_common_test_helpers:all(?MODULE).
+
+groups() ->
+    [].
+
+clickhouse_url() ->
+    erlang:iolist_to_binary([
+        <<"http://">>,
+        ?CLICKHOUSE_HOST,
+        ":",
+        erlang:integer_to_list(?CLICKHOUSE_DEFAULT_PORT)
+    ]).
+
+init_per_suite(Config) ->
+    case
+        emqx_common_test_helpers:is_tcp_server_available(?CLICKHOUSE_HOST, ?CLICKHOUSE_DEFAULT_PORT)
+    of
+        true ->
+            ok = emqx_common_test_helpers:start_apps([emqx_conf]),
+            ok = emqx_connector_test_helpers:start_apps([emqx_resource]),
+            {ok, _} = application:ensure_all_started(emqx_connector),
+            {ok, _} = application:ensure_all_started(emqx_ee_connector),
+            %% Create the db table
+            {ok, Conn} =
+                clickhouse:start_link([
+                    {url, clickhouse_url()},
+                    {user, <<"default">>},
+                    {key, "public"},
+                    {pool, tmp_pool}
+                ]),
+            {ok, _, _} = clickhouse:query(Conn, <<"CREATE DATABASE IF NOT EXISTS mqtt">>, #{}),
+            Config;
```

Comment:
Nit: close the connection at some point?

Reply:
Fixed